libs/sysplugins/smarty_internal_cacheresource_file.php

```diff
@@ -196,8 +196,8 @@ public function clear(Smarty $smarty, $resource_name, $cache_id, $compile_id, $e
      */
     public function hasLock(Smarty $smarty, Smarty_Template_Cached $cached)
     {
-        clearstatcache(true, $cached->lock_id);
-        if (is_file($cached->lock_id)) {
+        clearstatcache(true, $cached->lock_id ?? '');
+        if (null !== $cached->lock_id && is_file($cached->lock_id)) {
```

Comment:
since you closed previous one, carry the discussion:
- https://github.com/smarty-php/smarty/pull/711#discussion_r786668634

Reply:
Shall we continue the discussion here? My last answer was:

> Personally I agree with your point, but IMO it's out of scope of this change. What the original author tried to do is "doing whatever php < 8.1 would do". So another pr should cope with the issue.

Reply:
I have no say of this, as I'm not the maintainer.

personally, I would carry the fix to separate PR and hope it gets merged sooner as the change is smaller.

ps: edit `#199` to be backticks as it links to bogus ticket, or replace with link to source